dbms/src/AggregateFunctions/AggregateFunctionSumMap.h

```diff
@@ -253,6 +258,47 @@ class AggregateFunctionSumMap final : public IAggregateFunctionDataHelper<
     }
 
     const char * getHeaderFilePath() const override { return __FILE__; }
+
+    virtual bool keepKey(const T & key) const = 0;
```

Comment:
Should not be virtual (as we use CRTP).
You can write `static_cast<const Derived &>(*this).keepKey(key)`

Reply:
Right !